### PR TITLE
fix(planner): count aggregation function in subquery

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

For scalar subquery, we'll convert it to single join.

Single join is similar to left outer join, if there isn't matched row in the right side, we'll add **NULL** value for the right side.

But for **count** aggregation function, NULL values should be 0.

Fixes #issue

### CODE REVIEW

File: query/src/sql/optimizer/heuristic/subquery_rewriter.rs [hunk 1]
```diff
@@ -293,16 +306,44 @@ impl SubqueryRewriter {
...
+                        table_name: None,
+                        column_name: name,
+                        index,
+                        data_type,
+                        visible_in_unqualified_wildcard: false,
+                    },
+                });
+
+                let scalar = if flatten_info.from_count_func {
+                    // convert count aggregate function to multi_if function, if count() is null, then 0 else count()
```
Comment (leiysky) [inline query/src/sql/optimizer/heuristic/subquery_rewriter.rs:321]: Why do we need to rewrite `COUNT(*)` into `CASE WHEN COUNT(*) IS NULL THEN 0 ELSE COUNT(*) END`?

AFAIK, `COUNT(*)` will never return a `NULL` value.

Comment (xudong963) [reply query/src/sql/optimizer/heuristic/subquery_rewriter.rs:321]: Single join is similar to left outer join, if there isn't matched row on the right side, we'll add NULL value for the right side.  At the time, count(*) will be a NULL value.

File: tests/logictest/suites/query/subquery.test [hunk 2]
```diff
@@ -583,7 +583,44 @@ ORDER BY c.c_id, o.o_id
 5  NULL  NULL
 6  90    WA
 
+onlyif mysql
+statement query II
+SELECT
```
Comment (xudong963) [inline tests/logictest/suites/query/subquery.test:588]: You can consider the case, before the PR, the result is 
```sql
----
1  3
2  3
3  NULL
4  2
5  NULL
6  1

File: query/src/sql/optimizer/heuristic/subquery_rewriter.rs (hunk 1 above)
Comment (leiysky) [reply query/src/sql/optimizer/heuristic/subquery_rewriter.rs:321]: Ok, I get it.

So the `CASE WHEN` is actually applied to the column produced by single join  but not the `COUNT(*)`. 

And conincidentally, different from other set functions, `COUNT(*)` won't return NULL on empty input so it cannot utilize with the NULL produced by a single join as other aggregate functions.

Is that correct?

Comment (xudong963) [reply query/src/sql/optimizer/heuristic/subquery_rewriter.rs:321]: Yes, you totally got it!

**Review by leiysky — APPROVED** [review]

**Pull request merged**